### ISSUE TITLE
Sorting items on authors page broken by CSP

### ISSUE THREAD

**Issue by kentfredric** [metacpan/metacpan-web#2332]
On: https://metacpan.org/author/PEVANS

None of the top column sort mechanisms work.

Browser says:

```
01:49:57.385 Content Security Policy: The page’s settings blocked the loading of a resource at self (“script-src https://metacpan.org 'unsafe-inline' https://*.metacpan.org https://*.google-analytics.com https://*.google.com https://*.flattr.com”). Source: call to eval() or related function blocked by CSP. 1 dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:64
```
and
```
01:49:57.655 Error: call to eval() blocked by CSP 1 dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:64:1
	multisort https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:64:1
	.tablesorter</this.construct/</< https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:67:892
	dispatch https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:43038
	add/v.handle https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:41057
	trigger https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:71511
	trigger/< https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:72100
	each https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:2974
	each https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:1452
	trigger https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:72079
	.tablesorter</this.construct/< https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:68:30
	each https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:2974
	each https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:1452
	.tablesorter</this.construct https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:65:1619
	<anonymous> https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:651:24
	each https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:2974
	each https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:1452
	<anonymous> https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:645:420
	e https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:30003
	l/</t< https://metacpan.org/_assets/dcb3f4bdcf2ec22254eef2f95f9ed6c4778881b3.js:1:30307

```

Adblocking and anti-tracking things may be to blame. But this really shouldn't be a problem.

**Comment by kentfredric** [comment]
Probably broken by #2327

**Comment by kentfredric** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src

Eval is broken by the CSP, and it needs to declare *all* sources `unsafe-eval`  or *none* of them can use `eval`. 

https://stackoverflow.com/a/10949133/15614

The only workaround other than declaring `unsafe-eval` is to remove the need to `eval` entirely.

**Comment by mohawk2** [comment]
Seen. My starting point will be to eliminate the eval, else temporarily expand the CSP.

**Comment by kentfredric** [comment]
```javascript
dynamicExp+="return 0; ";dynamicExp+="}; ";if(table.config.debug){benchmark("Evaling expression:"+dynamicExp,new Date());}

eval(dynamicExp);cache.normalized.sort(sortWrapper);if(table.config.debug){benchmark("Sorting on "+sortList.toString()+" and dir "+order+" time:",sortTime);}
```

**Comment by kentfredric** [comment]
https://github.com/metacpan/metacpan-web/blob/f6d9725ecae678a8d8260f3504fd1f2e3b36d990/root/static/js/jquery.tablesorter.js#L586-L625

**Comment by kentfredric** [comment]
https://github.com/christianbach/tablesorter/issues/93

**Comment by kentfredric** [comment]
Confirming fixed (for now!) by #2333
